dagstore_test.go

```diff
@@ -456,49 +493,43 @@ func acquireShardThenRelease(t *testing.T, dagst *DAGStore, k shard.Key, n int)
 				return err
 			}
 
-			mu.Lock()
-			accessors = append(accessors, res.Accessor)
-			mu.Unlock()
+			accessors[i] = res.Accessor
 			return nil
 		})
 	}
 
 	require.NoError(t, grp.Wait())
-	info := flushAndGetShardState(t, dagst, k)
+
+	// check shard state.
+	info, err := dagst.GetShardInfo(k)
+	require.NoError(t, err)
 	require.Equal(t, ShardStateServing, info.ShardState)
 	require.NoError(t, info.Error)
 	// refs should be equal to number of acquirers since we've not closed any acquirer/released any shard.
 	require.EqualValues(t, n, info.refs)
-	// ensure transient file path exists.
-	abs := filepath.Join(dagst.config.TransientsDir, k.String())
-	_, err := os.Stat(abs)
-	require.NoError(t, err)
 
-	// close all the accessors
-	grp2, _ := errgroup.WithContext(context.Background())
-	for i := 0; i < n; i++ {
-		ac := accessors[i]
-		grp2.Go(ac.Close)
-	}
-
-	require.NoError(t, grp2.Wait())
-	info = flushAndGetShardState(t, dagst, k)
-	require.Equal(t, ShardStateAvailable, info.ShardState)
-	require.NoError(t, info.Error)
-	// refs should be zero now since shard accessors have been closed and transient file should be cleaned up.
-	require.Zero(t, info.refs)
-	_, err = os.Stat(abs)
-	require.Error(t, err)
+	return accessors
 }
 
-func flushAndGetShardState(t *testing.T, dagst *DAGStore, key shard.Key) ShardInfo {
-	ch := make(chan ShardResult, 1)
-	// wait for all existing ops for the shard to be processed by the event loop.
-	require.NoError(t, dagst.flush(context.Background(), key, ch))
-	<-ch
-	info, err := dagst.GetShardInfo(key)
-	require.NoError(t, err)
-	return info
+// releaseAll releases all accessors for a given shard.
+func releaseAll(t *testing.T, dagst *DAGStore, k shard.Key, accs []*ShardAccessor) {
+	grp, _ := errgroup.WithContext(context.Background())
+	for _, acc := range accs {
+		// close all accessors.
+		grp.Go(acc.Close)
```

Comment:
Shouldn't we not pass a looping variable to a go-routine ?

Reply:
Nope, because the usage of acc is not deferred, it's used immediately to obtain the reference to Close.